python/ray/tune/trial_runner.py

```diff
@@ -269,36 +369,27 @@ def checkpoint(self, force=False):
         Args:
             force (bool): Forces a checkpoint despite checkpoint_period.
         """
-        if not self._local_checkpoint_dir:
-            return
-        now = time.time()
-        if now - self._last_checkpoint_time < self._checkpoint_period and (
-                not force):
-            return
-        self._last_checkpoint_time = now
-        runner_state = {
-            "checkpoints": list(
-                self.trial_executor.get_checkpoints().values()),
-            "runner_data": self.__getstate__(),
-            "stats": {
-                "start_time": self._start_time,
-                "timestamp": self._last_checkpoint_time
-            }
-        }
-        tmp_file_name = os.path.join(self._local_checkpoint_dir,
-                                     ".tmp_checkpoint")
-        with open(tmp_file_name, "w") as f:
-            json.dump(runner_state, f, indent=2, cls=TuneFunctionEncoder)
-
-        os.replace(tmp_file_name, self.checkpoint_file)
-        self._search_alg.save_to_dir(
-            self._local_checkpoint_dir, session_str=self._session_str)
-
-        if force:
-            self._syncer.sync_up()
-        else:
-            self._syncer.sync_up_if_needed()
-        return self._local_checkpoint_dir
+        context = None
+        if not self._checkpoint_manager.auto_checkpoint_period:
+            context = warn_if_slow(
+                "experiment_checkpoint",
+                message="Checkpointing the experiment state took "
+                "{duration:.3f} s, which may be a performance "
+                "bottleneck. Please ensure the "
+                "`TUNE_GLOBAL_CHECKPOINT_S` environment variable is "
+                "something significantly higher than this duration "
+                "to ensure compute time is mostly spent on the main "
+                "training loop.").__enter__()
```

Comment:
instead, how about do a `warn_if_slow(msg, disable=self._checkpoint_manager.auto_checkpoint_period)` so that we don't need to do the `__enter/exit__`?